api/app/signals/apps/questionnaires/tests/services/test_answer.py

```diff
@@ -104,3 +104,30 @@ def test_validate_field_type_select_object(self):
 
         payload_not_on_map['on-map'] = False
         validate_answer(payload_not_on_map, selected_object_question)
+
+        selected_object_question.multiple_answers_allowed = True
+        selected_object_question.save()
+
+        validate_answer([payload_not_on_map for x in range(3)], selected_object_question)
+
+    def test_validate_multiple_answers(self):
+        q = QuestionFactory.create(field_type='plain_text', multiple_answers_allowed=True, required=True,
+                                   extra_properties={'answers': {'minItems': 2, 'maxItems': 5}})
+
+        AnswerService.validate_answer_payload(['answer 1', 'answer 2'], q)
+        AnswerService.validate_answer_payload([f'answer {x}' for x in range(5)], q)
+        with self.assertRaises(django_validation_error):
+            # Not a list
+            AnswerService.validate_answer_payload('answer is not a list', q)
+
+            # Only one answer is given, the minimum is 2
+            AnswerService.validate_answer_payload(['answer 1', ], q)
+
+            # An empty list
+            AnswerService.validate_answer_payload([], q)
+
+            # Not strings
+            AnswerService.validate_answer_payload([1, 2, 3], q)
+
+            # Too much answers
+            AnswerService.validate_answer_payload([f'answer {x}' for x in range(6)], q)
```

Comment:
This test fails in that it does not raise a validation error. This is masked because the this test does not run as it is preceded by several calls to `AnswerService.validate_answer_payload` that (would) raise a ValidationError in the same `with self.assertRaises(django_validation_error)` block.

The reason `AnswerService.validate_answer_payload([f'answer {x}' for x in range(6)], q)` does not fail is the mixup in api/app/signals/apps/questionnaires/models/question.py (mentioned in a comment for that file).

To fix this test implement the suggestion for the Question model and change `extra_properties={'answers': {'minItems': 2, 'maxItems': 5}})` to `additional_validation={'answers': {'minItems': 2, 'maxItems': 5}})` in line 115 in this test.

Reply:
I guess you where writing this review before I got the change to rebase and fix some of the issues. This has been done now